packages/core/lib/create.js

```diff
@@ -55,7 +46,7 @@ var toUnderscoreFromCamel = function(string) {
 };
 
 var Create = {
-  contract: function(directory, name, options, callback) {
+  contract: async function(directory, name, options, callback) {
```

Comment:
The `await` keyword causes this function to return a new `Promise`. It's possible `Create.contract` isn't used in a context where this could cause confusion, but if it is exposed as a public API it would be a little strange.

Reply:
Ooh, I missed that this is exposed as a library in this package and so it would probably need to have a dual Promise/callback interface. In version 6 of Truffle all this stuff should probably be fully redone to just use Promises. Is there any reason to keep its callback api as well?

Reply:
A reason to keep the callback style in version 6 would be to avoid the annoyance of breaking changes. I prefer keeping it for one major version while also shipping a deprecation notice if that style is used (e.g., `console.warn("Callback style for {package/method name} is deprecated and will be removed in v7");`). Of course, this must makes it annoying for us as the developers of truffle.